Add Collection Controller Documentation (#37)
kuzzleio/sdk-csharp: doc/1/controllers/collection/create/index.md

---
code: true
type: page
title: CreateAsync
description: Create a new collection.
---

# CreateAsync

Creates a new [collection](/core/1/guides/essentials/store-access-data) in the provided `index`.

> You can also provide an optional data mapping that allow you to exploit the full capabilities of our

Reviewer: You should use the API description:

```suggestion
You can also provide an optional body with a [collection mapping](/core/1/guides/essentials/database-mappings) allowing you to exploit the full capabilities of our persistent data storage layer.
```

> persistent data storage layer, [ElasticSearch](https://www.elastic.co/products/elasticsearch) (check here the [mapping capabilities of ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/5.4/mapping.html)).

Reviewer: ```suggestion
```

This method will only update the mapping if the collection already exists.

## Arguments

```csharp
public async Task CreateAsync(
        string index,
        string collection,
        JObject mappings = null);
```

| Argument     | Type                           | Description                                      |
|--------------|--------------------------------|--------------------------------------------------|

> | `index`      | <pre>string&</pre>             | Index name                                       |

Reviewer: ```suggestion
| `index`      | <pre>string</pre>             | Index name                                       |
```

> | `collection` | <pre>string&</pre>             | Collection name                                  |

Reviewer: ```suggestion
| `collection` | <pre>string</pre>             | Collection name                                  |
```

| `mapping`    | <pre>JObject</pre><br>(`null`) | JObject representing the collection data mapping |

### mapping

A JObject representing the collection data mapping.

> The mapping must have a root field `properties` that contain the mapping definition:

Reviewer: ```suggestion
The mapping must have a root field `properties` containing the mapping definition:
```

Reviewer: Also, that description is now obsolete. See: https://docs.kuzzle.io/core/1/api/controllers/collection/create/#body-properties

```json
{
  "properties": {
    "field1": { "type": "text" },
    "field2": {
      "properties": {
        "nestedField": { "type": "keyword" }
      }
    }
  }
}
```

> More informations about database mappings [here](/core/1/guides/essentials/database-mappings).

Reviewer: "information" is non-countable:

```suggestion
More information about database mappings [here](/core/1/guides/essentials/database-mappings).
```
